pkg/3scale/amp/operator/redis_reconciler.go

```diff
@@ -124,3 +147,100 @@ func Redis(apimanager *appsv1alpha1.APIManager, client client.Client) (*componen
 	}
 	return component.NewRedis(opts), nil
 }
+
+func (r *RedisReconciler) deleteSystemRedisSecretNamespaceAttribute() (reconcile.Result, error) {
+	redis, err := Redis(r.apiManager, r.Client())
+	if err != nil {
+		return reconcile.Result{}, err
+	}
+
+	desired := redis.SystemRedisSecret()
+	existing := &corev1.Secret{}
+	err = r.Client().Get(context.TODO(), types.NamespacedName{Name: desired.Name, Namespace: r.apiManager.Namespace}, existing)
+	if k8serr.IsNotFound(err) {
```

Comment:
I don't think we can ever land here, is it fair to say that at this point, we are pretty sure we are not going to get "isNotFound" error since it would have failed earlier? Can you confirm this is really required?

Reply:
I had the problem, so I placed it. I can retest and confirm if required.

Reply:
hmm..please re-check if you can. It seems a bit late to be checking if the secret is present because we already created deployments that reference the secret (in envs)

Reply:
Thanks, just checked it too, so we do create secret afterwards. Thanks